### PR TITLE
Refactor Credentials/Authorization

### PR DESCRIPTION
**Pull request by ianwremmel**

### CODE REVIEW

File: packages/node_modules/@ciscospark/spark-core/test/integration/spec/credentials/credentials.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+describe(`spark-core`, () => {
+  describe(`Credentials`, () => {
+    let user;
+    before(() => testUsers.create({count: 1})
+      .then(([u]) => {
+        user = u;
+      }));
+
+    describe(`#refresh()`, () => {
+      nodeOnly(it)(`refreshes an access token`, () => {
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/spark-core/test/integration/spec/credentials/credentials.js]: What does `it` refer to?

Comment (adamweeks) [inline packages/node_modules/@ciscospark/spark-core/test/integration/spec/credentials/credentials.js]: Nevermind. I think I got it now.

File: packages/node_modules/@ciscospark/common/src/in-browser/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**!
+ *
+ * Copyright (c) 2015-2017 Cisco Systems, Inc. See LICENSE file.
+ * @ignore
+ */
+
+/**
+ * Constants that, through the magic of bundling should indicate if we're
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/common/src/in-browser/index.js]: magic

File: packages/node_modules/@ciscospark/common/src/make-state-datatype.js [hunk 3]
```diff
@@ -15,6 +15,7 @@
 //     dataType.set = dataType.set.bind(this);
 //   }
 // }.bind(this));
+// // END HACK
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/common/src/make-state-datatype.js]: I prefer `</HACK>` but this should work. 😝

File: packages/node_modules/@ciscospark/spark-core/test/unit/spec/credentials/token.js [hunk 4]
```diff
@@ -207,6 +207,18 @@ describe(`spark-core`, () => {
...
+        it(`accepts a string as an access token`, () => {
+          const token = new Token(`AT`, {parent: spark});
+          assert.isTrue(token.canAuthorize);
+          assert.equal(token.toString(), `Bearer AT`);
+        });
+
+        it(`accepts a string as an access token (with token type)`, () => {
+          const token = new Token(`Fake AT`, {parent: spark});
+          assert.isTrue(token.canAuthorize);
+          assert.equal(token.toString(), `Fake AT`);
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/spark-core/test/unit/spec/credentials/token.js]: How does this pass? I'm not seeing where it parses out the token type.

File: packages/node_modules/@ciscospark/spark-core/test/integration/spec/credentials/credentials.js (hunk 1 above)
Comment (ianwremmel) [inline packages/node_modules/@ciscospark/spark-core/test/integration/spec/credentials/credentials.js]: `Credentials#refresh()`

File: packages/node_modules/@ciscospark/spark-core/test/unit/spec/credentials/token.js (hunk 4 above)
Comment (ianwremmel) [inline packages/node_modules/@ciscospark/spark-core/test/unit/spec/credentials/token.js]: It's in `Token#set()`

**Pull request merged**